presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -228,6 +231,32 @@ public static Boolean stIsEmpty(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return deserialize(input).isEmpty();
     }
 
+    @Description("Returns true if the input geometry is well formed")
+    @ScalarFunction("ST_IsValid")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean stIsValid(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
+    {
+        return GeometryJtsUtils.deserialize(input).isValid();
+    }
+
+    @Description("If input geometry is not valid, return a reason why. Otherwise, returns null.")
+    @ScalarFunction("geometry_is_valid_reason")
+    @SqlType(StandardTypes.VARCHAR)
+    @SqlNullable
+    public static Slice isValidReason(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
+    {
+        if (input == null) {
+            return null;
+        }
+        Geometry geometry = GeometryJtsUtils.deserialize(input);
+        IsValidOp isValidOp = new IsValidOp(geometry);
```

Comment:
`isValid` does a validation under the hood with `checkValid`. Then we do another validation with `getValidationError` below. We can get away with a single validation:
```
    TopologyValidationError error = new IsValidOp(geometry).getValidationError(); // got rid of the isValidOp local variable
    if (error == null) {
        return null;
    }
    return utf8Slice(error.toString()); // added static import
```